agntcy_acp/langgraph/acp_node.py

```diff
@@ -192,27 +199,114 @@ def _handle_run_output(self, state: Any, run_output: RunOutput):
             run_error: RunError = run_output.actual_instance
             raise ACPRunException(f"Run Failed: {run_error}")
         elif isinstance(run_output.actual_instance, RunInterrupt):
-            raise ACPRunException(f"ACP Server returned a unsupporteed interrupt response: {run_output}")
+            # This case is handled in the invokes
+            pass
+
         else:
-            raise ACPRunException(f"ACP Server returned a unsupporteed response: {run_output}")
+            raise ACPRunException(
+                f"ACP Server returned a unsupporteed response: {run_output}"
+            )
 
         return state
 
-    def invoke(self, state: Any, config: RunnableConfig) -> Any:
+    def _get_thread_id(self, config):
+        configurable = config.get("configurable", {})
+        thread_id = configurable.get("thread_id", None)
+        return thread_id
+
+    def _handle_interrupt(self, config, run_output):
+        thread_id = self._get_thread_id(config=config)
```

Comment:
When you call _handle_interrupt, you should always have a thread_id in your hands, why don't you pass it instead of getting it again?